force-app/main/default/classes/DMLOperations.cls

```diff
@@ -198,10 +274,15 @@ public with sharing class DMLOperations {
      */
     public static void insertAndDeleteLeads(List<String> leadNames) {
         // Loop through each Lead name
-
+        List <Lead> insertLeads = new List<Lead>();
+          for (String lead : leadNames) {
```

Comment:
For clarity, a better variable name here would be `name` or `leadName`. Save `lead` for the object Lead.